lib/cocoapods/generate/configuration.rb

```diff
@@ -303,12 +304,13 @@ def to_s
         end << ' }'
       end
 
-      # @return [Pathname] the directory for installation of the generated workspace
+      # @return [Pathname] the directory for installation of the generated workspace.
       #
-      # @param  [String] name the name of the pod
+      # @param  [Array<Specification>] specs The specs to get a directory name for.
       #
-      def gen_dir_for_pod(name)
-        gen_directory.join(name)
+      def gen_dir_for_specs(specs)
+        basename = specs.count == 1 ? specs.first.name : 'Workspace'
```

Comment:
not sure if there's docs on this anywhere but calling this out - if using a single workspace for multiple pods, the generated workspace will now be in `gen/Workspace` since there isn't a single pod from which to determine a name